src/app/ui/dialog/dialog-container/dialog-container.component.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  Component,
+  OnInit,
+  ChangeDetectionStrategy,
+  HostListener,
+  ElementRef,
+  Output,
+  EventEmitter,
+  Renderer,
+  AfterViewInit,
+  Input
+} from '@angular/core';
+
+export interface DialogOptions {
+  closeOnEsc?: boolean;
+  closeOnClickOutside?: boolean;
+}
+
+@Component({
+  selector: 'iw-dialog-container',
+  templateUrl: './dialog-container.component.html',
+  styleUrls: ['./dialog-container.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DialogContainerComponent implements OnInit, AfterViewInit {
+  @Input() dialogOptions: DialogOptions;
+  @Output() onClose = new EventEmitter();
+
+  @HostListener('window:keydown', ['$event'])
+  onKeydown($event: KeyboardEvent) {
+    if ($event.keyCode === 27 && this.dialogOptions.closeOnEsc) {
```

Comment:
You told me that `keyCode` is deprecated. Is there no good alternative? What about `$event.key`?